### PR TITLE
Add language Skript

### PR DESCRIPTION
**Pull request by BlockArchitech**
## Description

This will add syntax highlighting, and language identification of the language Skript across GitHub.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension: https://github.com/search?p=2&q=extension%3Ask&type=Code
      
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s): https://github.com/BedrockMiners/Skript-Repository
      - [URL to each sample source, if applicable]
    - Sample license(s): https://github.com/BedrockMiners/Skript-Repository/blob/main/LICENSE
  
  - [x] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/?utf8=✓&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2FSkript-Atom%2FLanguage-Skript%2Fblob%2Fmaster%2Fgrammars%2Fskript.cson&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2FBedrockMiners%2FSkript-Repository%2Fblob%2Fmain%2Fcore.sk&code=
  - [x] I have updated the heuristics to distinguish my language from others using the same extension. - Not Needed, .sk is unique

### CODE REVIEW

**Comment by BlockArchitech** [comment]
* Should note BedrockMiners is an old project I was working on. Also, some samples created by me were added to the /samples Directory.

**Comment by BlockArchitech** [comment]
Also, adding on to the color, this is the official color for the Script language, which is plastered all over the Documentation.
[Skript Docs](https://skriptlang.github.io/Skript/)

File: samples/Skript/command-hello.sk
Comment (Nixinova) [inline samples/Skript/command-hello.sk:6]: The samples should be more realistic/representative of the language. For instance these are very basic and are missing some of the syntax shown on the website like {} and %%.

Comment (BlockArchitech) [reply samples/Skript/command-hello.sk:6]: I'll fix these in the morning.

**Comment by BlockArchitech** [comment]
Alright @Nixinova , I added the new sample files.

**Comment by BlockArchitech** [comment]
Hey, @Nixinova can I get a follow up?

**Comment by Nixinova** [comment]
Those new samples should be good. You'll have to update the OP listing where they're from tho if they are taken from somewhere else.

**Comment by BlockArchitech** [comment]
Awesome!

**Comment by BlockArchitech** [comment]
Hey @Alhadis can you review this?

**Comment by Alhadis** [comment]
@BlockArchitech I'll need to review the distribution of `.sk` files across GitHub to ascertain that Skript meets our criteria for popularity. 57,658 search results for the `.sk` extension looks promising, but not all of those files are guaranteed to be Skript files. In particular, `sk` might be used by files written in Slovakian — it's common to see files suffixed with their language's [BCP-47 tag](https://en.wikipedia.org/wiki/IETF_language_tag); i.e., stuff like [this](https://github.com/apache/httpd/tree/00a3adb4283c1db943384d4877df0255d8121f0a/docs/manual/mod).

I'll get back to you.

**Comment by BlockArchitech** [comment]
So @Alhadis I've found [this](https://github.com/NCAR/aircraft_ads2_projects) which seems to be some aircraft thing which is WAYY out of my comfort zone when it comes to this. I'll have to find out how to create heuristics to differentiate Skript from aircraft controller code.

**Comment by BlockArchitech** [comment]
Reading the guidelines on heuristics, I have to run the Bayesian classifier, and the guidelines said to ping @lildude to help with this.

**Comment by Alhadis** [comment]
Heuristics are only necessary when there are multiple (eligible) languages with the same file extension(s) and/or filename(s). They're not necessary for every addition, and the part about benchmarking the classifier is only relevant for complex heuristics… I think.

The main issue I need to address is the possibility of as-yet undiscovered formats with the `.sk` extension that are _also_ eligible for addition to Linguist (though unlikely, it *has* happened before…)

In other words, don't worry about if for the time being. 😉

**Comment by BlockArchitech** [comment]
Hey, any update?

**Comment by BlockArchitech** [comment]
Okay so @Alhadis I went through most of the search results and found that a lot of the results were skript, but some weren't. I'll create an excel document with a chart to better visualize this tomorrow.

**Comment by BlockArchitech** [comment]
Also I'll merge the new version into this pr

**Comment by Alhadis** [comment]
You might want to hold off on that too, as there was a recent (temporary) change in Linguist's eligibility assessment. See #5756.

**Comment by BlockArchitech** [comment]
Okay, that's going to make my life a LOT easier even outside of linguist. Just signed up.

File: Gemfile
Comment (lildude) [inline Gemfile:4]: Huh? Please remove this.

**Review by lildude — CHANGES_REQUESTED** [review]
Looks like you missed the step about adding the grammar license file. When you ran `script/add-grammar` it should have downloaded the file in a format we need.

File: Gemfile
Comment (BlockArchitech) [reply Gemfile:4]: Ah, I have a program that automatically does this. I'll fix this and add the license file.

**Comment by Moderocky** [comment]
Hello, I'm here in an official capacity from SkriptLang, the org that maintains the Skript language.

We have some major concerns about this PR:
Both samples provided are invalid. `sample1.sk` contains several syntactical errors (incorrect quoting, illegal nesting of options, etc.) and `sample2.sk` relies on third-party modular addons that are not a default part of the Skript language.
The current samples seem to be inadequate when compared to other languages from the linguist repository. Very little of the feature-set is sampled, and proper explanatory comments ought to be provided.

It should also be noted that neither the highlighter grammar vendor linked nor the sample files are officially created/maintained - both are provided by third parties not affiliated the language maintainers.
(In fact, the Skript-Atom project has been abandoned for over five years.)

While we'd be honoured to have linguist support Skript, we want to make sure that the language is represented accurately and reliably.

**Comment by BlockArchitech** [comment]
> We have some major concerns about this PR: Both samples provided are invalid. `sample1.sk` contains several syntactical errors (incorrect quoting, illegal nesting of options, etc.) and `sample2.sk` relies on third-party modular addons that are not a default part of the Skript language. The current samples seem to be inadequate when compared to other languages from the linguist repository. Very little of the feature-set is sampled, and proper explanatory comments ought to be provided.

I'll address these sample files, for sample2 I did not know it relied on a 3rd-party module. sample1 I'm not sure I understand your claim as this was a tested file, but I can go ahead and make a new one. 


> It should also be noted that neither the highlighter grammar vendor linked nor the sample files are officially created/maintained - both are provided by third parties not affiliated the language maintainers.
(In fact, the Skript-Atom project has been abandoned for over five years.)

Grammar has been updated.

Thanks for your concern, and letting me know. I'll go ahead and get these added ASAP.

**Comment by BlockArchitech** [comment]
Quick Update: The folks over at SkriptLang (big thanks @Moderocky ) have been working on official grammar. On stable release I'll get this updated and re-request another review. [Repo](https://github.com/SkriptLang/skript-grammar)

**Comment by BlockArchitech** [comment]
Ah, please ignore the double-merge, did once on GitHub and `git pull` on local machine

**Pull request closed without merging**
